core/node/storage/pg_storage.go

```diff
@@ -1754,3 +1753,169 @@ func (s *PostgresEventStore) debugReadStreamData(
 
 	return result, nil
 }
+
+func (s *PostgresEventStore) StreamLastMiniBlock(
+	ctx context.Context,
+	streamID StreamId,
+) (*LatestMiniBlock, error) {
+	var ret *LatestMiniBlock
+	err := s.txRunner(
+		ctx,
+		"StreamLastMiniBlocks",
+		pgx.ReadOnly,
+		func(ctx context.Context, tx pgx.Tx) error {
+			var err error
+			ret, err = s.lastMiniBlockForStream(ctx, tx, streamID)
+			return err
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+func (s *PostgresEventStore) lastMiniBlockForStream(
+	ctx context.Context,
+	tx pgx.Tx,
+	streamID StreamId,
+) (*LatestMiniBlock, error) {
+	miniblockNumsRow, err := tx.Query(
+		ctx,
+		`select o.stream_id, o.seq_num, o.blockdata from miniblocks o inner join (
+			select stream_id, max(seq_num) as max_seq_num from miniblocks
+				WHERE stream_id = $1 group by stream_id
+			) i on o.stream_id = i.stream_id and o.seq_num = i.max_seq_num`,
+		streamID,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer miniblockNumsRow.Close()
+
+	if miniblockNumsRow.Next() {
+		var (
+			streamId  StreamId
+			maxSeqNum int64
+			blockData []byte
+		)
+
+		if err = miniblockNumsRow.Scan(&streamId, &maxSeqNum, &blockData); err != nil {
```

Comment:
To match suggestion above:
```suggestion
		if err = miniblockNumsRow.Scan(&maxSeqNum, &blockData); err != nil {
```